src/error.rs

```diff
@@ -0,0 +1,94 @@
+use serde::Serialize;
+use std::fmt;
+
+#[derive(thiserror::Error, Debug, Copy, Clone, Eq, PartialEq, Serialize)]
+pub enum ErrorKind {
+    #[error("DID not resolved.")]
+    DIDNotResolved,
```

Comment:
In Python it's DIDDocNotResolved, but probably it's minor, we can adjust later.

Reply:
agree